docs/src/extensions.md

```diff
@@ -1,5 +1,93 @@
+```@meta
+CurrentModule = JuMP
+```
+
 Extending JuMP
 ==============
 
 TODO: How to extend JuMP: discussion on different ways to build on top of JuMP.
 How to extend JuMP's macros and how to avoid doing this.
+
+## Extending MOI
+
+TODO: Create new MOI function/sets, how to use it in JuMP, create new bridge
+
+## Extending JuMP macros
+
+In order to provide the most convenient syntax for the user to create variables,
```

Comment:
"Most convenient" implies that we think that the macros are the best possible syntax. Maybe "a convenient syntax".